nomad/structs/structs.go

```diff
@@ -8437,6 +8437,35 @@ func (ts *TaskState) Successful() bool {
 	return ts.State == TaskStateDead && !ts.Failed
 }
 
+func (ts *TaskState) Equal(o *TaskState) bool {
+	if ts.State != o.State {
+		return false
+	}
+	if ts.Failed != o.Failed {
+		return false
+	}
+	if ts.Restarts != o.Restarts {
+		return false
+	}
+	if ts.LastRestart != o.LastRestart {
+		return false
+	}
+	if ts.StartedAt != o.StartedAt {
+		return false
+	}
+	if ts.FinishedAt != o.FinishedAt {
+		return false
+	}
+	if len(ts.Events) != len(o.Events) {
+		return false
+	}
+	if ts.TaskHandle != o.TaskHandle {
+		return false
+	}
```

Comment:
`Events` and `TaskHandle` are complex structs - should we not defer to `Equal()` implementation on each of them?

Reply:
Good call. Resolved in f62c33f7e4fc40db35ed3287d8089e6258a3fe58